packages/crayons-core/src/components/datepicker/datepicker.tsx

```diff
@@ -875,6 +833,115 @@ export class Datepicker {
     }
   }
 
+  processRangeValueChange(val, emitChange = false) {
+    let [fromDate, toDate] =
+      val?.split(TranslationController.t('datepicker.to')) || [];
+    fromDate = fromDate?.trim();
+    toDate = toDate?.trim();
+
+    const parsedFromDate = parse(fromDate, this.displayFormat, new Date(), {
+      locale: this.langModule,
+    });
+
+    const parsedToDate = parse(toDate, this.displayFormat, new Date(), {
+      locale: this.langModule,
+    });
+
+    const isValidFromDate = isValid(parsedFromDate);
+    const isValidToDate = isValid(parsedToDate);
+
+    const year = getYear(
+      parse(fromDate, this.displayFormat, new Date(), {
+        locale: this.langModule,
+      })
+    );
+    const toYear = getYear(
+      parse(toDate, this.displayFormat, new Date(), {
+        locale: this.langModule,
+      })
+    );
+    if (
+      !isValidFromDate ||
+      !isValidToDate ||
+      !isMatch(fromDate, this.displayFormat, {
+        locale: this.langModule,
+      }) ||
+      !isMatch(toDate, this.displayFormat, {
+        locale: this.langModule,
+      }) ||
+      year < this.minYear ||
+      year > this.maxYear ||
+      toYear < this.minYear ||
+      toYear > this.maxYear ||
+      !this.isDatewithinRange(parsedFromDate.valueOf(), parsedToDate.valueOf())
+    ) {
+      // Invalid date format
+      this.isDateInvalid = !!val && true;
```

Comment:
please remove `&& true` . `!!val` is enough to convert it in to a boolean value.

Reply:
fixed